activerecord/lib/active_record/schema_dumper.rb

```diff
@@ -243,7 +243,9 @@ def format_options(options)
       end
 
       def remove_prefix_and_suffix(table)
-        table.gsub(/^(#{@options[:table_name_prefix]})(.+)(#{@options[:table_name_suffix]})$/,  "\\2")
+        prefix = Regexp.escape(@options[:table_name_prefix].to_s)
+        suffix = Regexp.escape(@options[:table_name_suffix].to_s)
+        table.sub(/\A(?:#{prefix})(.+)(?:#{suffix})\z/, "\\1")
```

Comment:
I didn't realized that we could remove the capturing groups.
Original code was `table.gsub(/^(#{...})(.+)(#{...})$/,  "\\2")`, I was only avoided the unused capturing (first and third `(...)` to `(?:...)`). As @lugray said, I also think that the unused capturing could be removed completely.